src/libraries/System.Runtime/tests/System.Runtime.Tests/System/ActivatorTests.cs

```diff
@@ -107,6 +107,65 @@ public void CreateInstance_PublicOnlyTypeWithPrivateDefaultConstructor_ThrowsMis
             Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(TypeWithPrivateDefaultConstructor), nonPublic: false));
         }
 
+        [Fact]
+        public void CreateInstance_WithCustomBinder_ThrowsMissingMethodException()
+        {
+            // MissingMethodException not caused by a binder must not contain an inner exception.
+            var mme = Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(TypeWithPrivateDefaultConstructor), nonPublic: false));
+            Assert.Contains("System.Tests.ActivatorTests.TypeWithPrivateDefaultConstructor", mme.Message);
+            Assert.Null(mme.InnerException);
+
+            // MissingMethodException caused by a binder must be wrapped.
+            mme = Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(
+                typeof(object), BindingFlags.CreateInstance,
+                new CustomBinder() { BindToMethodAction = () => throw new MissingMethodException("Hello, World!!") },
+                null, null, null
+            ));
+            Assert.Contains("System.Object", mme.Message);
+            Assert.NotNull(mme.InnerException);
+            Assert.IsType<MissingMethodException>(mme.InnerException);
+            Assert.StrictEqual("Hello, World!!", mme.InnerException.Message);
+
+            // Any other exceptions will not be caught.
+            Assert.Throws<Exception>(() => Activator.CreateInstance(
+                typeof(object), BindingFlags.CreateInstance,
+                new CustomBinder() { BindToMethodAction = () => throw new Exception() },
+                null, null, null
+            ));
+
+            // MissingMethodException must not contain an inner exception when BindToMethod returns null.
+            mme = Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(
+                typeof(object), BindingFlags.CreateInstance,
+                new CustomBinder() { BindToMethodAction = () => null },
+                null, null, null
+            ));
+            Assert.Contains("System.Object", mme.Message);
+            Assert.Null(mme.InnerException);
+        }
+
+        class CustomBinder : Binder
+        {
+            public required Func<MethodBase?> BindToMethodAction { get; init; }
+
+            public override MethodBase BindToMethod(BindingFlags bindingAttr, MethodBase[] match, ref object?[] args, ParameterModifier[]? modifiers, CultureInfo? culture, string[]? names, out object? state)
```

Comment:
To avoid build errors:

```suggestion
            public override MethodBase BindToMethod(BindingFlags bindingAttr, MethodBase[] match, ref object?[] args, ParameterModifier[]? modifiers, CultureInfo? culture, string[]? names, out object? _)
```